pkg/api/openshiftcluster.go

```diff
@@ -258,12 +258,13 @@ type OIDCIssuer string
 type ClusterProfile struct {
 	MissingFields
 
-	PullSecret           SecureString         `json:"pullSecret,omitempty"`
-	Domain               string               `json:"domain,omitempty"`
-	Version              string               `json:"version,omitempty"`
-	ResourceGroupID      string               `json:"resourceGroupId,omitempty"`
-	FipsValidatedModules FipsValidatedModules `json:"fipsValidatedModules,omitempty"`
-	OIDCIssuer           OIDCIssuer           `json:"oidcIssuer,omitempty"`
+	PullSecret                    SecureString         `json:"pullSecret,omitempty"`
+	Domain                        string               `json:"domain,omitempty"`
+	Version                       string               `json:"version,omitempty"`
+	ResourceGroupID               string               `json:"resourceGroupId,omitempty"`
+	FipsValidatedModules          FipsValidatedModules `json:"fipsValidatedModules,omitempty"`
+	OIDCIssuer                    OIDCIssuer           `json:"oidcIssuer,omitempty"`
```

Comment:
should this be a pointer?

Reply:
This comes from the changes proposed in [Preview API design doc](https://docs.google.com/document/d/1AIuf6H9y0Jjboywp0NgD8-9lU1jCj5WrE83jh58rEg4/edit#heading=h.cifwpl8od05m), and were pushed as part of [Preview API fields addition PR](https://github.com/Azure/ARO-RP/pull/3478/files#diff-c7bfe4dfde8a1a575263edc226847cbab8ac7f60f04193f531a17d73571a3398R81).
Not sure why this field need to be a pointer. The MIWI cluster check would be done on ServicePrincipalProfile field which is changed to pointer.

Reply:
In a recent code review did we decide to make this a pointer after all just for the purpose of being able to make it nil, or am I getting this field confused with `BoundServiceAccountSigningKey` just below?

Reply:
`OIDCIssuer` has a similar behavior as `BoundServiceAccountSigningKey` and will always be set when it is a MIWI cluster. Otherwise not set.
@kimorris27 can confirm if it should be a pointer or not, as he is the owner of  [Preview API design doc](https://docs.google.com/document/d/1AIuf6H9y0Jjboywp0NgD8-9lU1jCj5WrE83jh58rEg4/edit#heading=h.cifwpl8od05m). Also the `BoundServiceAccountSigningKey` has been changed to *SecureString.

Reply:
I don't see why either of these fields (`OIDCIssuer` and `BoundServiceAccountSigningKey`) _must_ be pointers. I feel like the difference is mostly semantic. However, in order to be consistent with how things are done in existing code, I would make both of these fields values rather than pointers if it were up to me.

For example, `ClusterProfile.PullSecret` is optional, but it's a value rather than a pointer.

I'd defer to @bennerv though since he raised the question in the first place.

Reply:
I felt the same. Approving either way.

Reply:
After some discussion, it seems that since the `OIDCIssuer` being a non-pointer will be visible to the customer when doing `az aro show`. So it has been decided to convert `OIDCIssuer` & `BoundServiceAccountSigningKey` to pointer.
Commit Ref:-  https://github.com/Azure/ARO-RP/pull/3564/commits/2d223560cd9435dea2202a462bfddb63cfd71b13

Reply:
That choice makes sense to me 👍🏻

Reply:
Resolving this thread since the change has been made and the motivation for the change is clear.